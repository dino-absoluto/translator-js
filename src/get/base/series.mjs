/**
 * @file Get module
 * @author Dino <dinoabsoluto+dev@gmail.com>
 * @license
 * This file is part of translator-js.
 *
 * translator-js is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * translator-js is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with translator-js.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
/* imports */
import Base from './base'
import path from 'path'
import fs from 'fs'
/* -imports */

export default class Series extends Base {
  constructor (props) {
    const meta = Series.parseMeta(props)
    super(meta)
    Object.defineProperties(this, {
      url: { enumerable: true, get: () => this.props.url }
    })
  }

  static parseMeta (props) {
    try {
      let url = new URL(props.source)
      return {
        url
      }
    } catch (err) {
      try {
        let fname = path.join(props.source, 'index.json')
        let data = JSON.parse(fs.readFileSync(fname, 'utf8'))
        data.url = new URL(data.url)
        return data
      } catch (error) {
        throw new Error('Failed to read index.json')
      }
    }
  }

  refresh () {
  }
}
