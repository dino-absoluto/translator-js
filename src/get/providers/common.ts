/**
 * @file Common code
 * @author Dino <dinoabsoluto+dev@gmail.com>
 * @license
 * This file is part of translator-js.
 *
 * translator-js is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * translator-js is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with translator-js.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
/* imports */
/* code */

export type ContextCallback = (name: string) => Buffer | string[]
export abstract class Context {
  abstract requestFile (
    name: string,
    fn: ContextCallback): void

  parseNode (node: Node): string {
    return node.textContent || ''
  }
}

export type Content = (fmt: Context) => void

export interface ChapterData {
  group?: string
  name: string
  updateId?: string
  content?: Content
}

export interface Chapter extends ChapterData {
  fetch (): Promise<void>
}

export interface NovelData {
  readonly id: string
  name?: string
  author?: string
  description?: string
  keywords?: string[]
  genre?: string
  status?: {
    completed: boolean
    size: number
  }
}

export interface Novel extends NovelData {
  fetch (): Promise<void>
  fetchIndex (): Promise<Chapter[]>
}

export interface Provider {
  readonly acceptDomains: string[]
  fromURL (href: URL): Novel | undefined
}
