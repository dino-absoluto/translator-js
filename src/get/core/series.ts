/**
 * @file Series
 * @author Dino <dinoabsoluto+dev@gmail.com>
 * @license
 * This file is part of translator-js.
 *
 * translator-js is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * translator-js is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with translator-js.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
/* imports */
import { Novel, NovelData } from '../providers'
/* code */

export interface SeriesData {
  readonly novel?: NovelData
}

export class Series implements SeriesData {
  readonly novel: Novel
  constructor (novel: Novel, data?: NovelData) {
    if (data) {
      if (data.id !== novel.id) {
        throw new Error('Novel and data don\'t match')
      }
      Object.assign(novel, data)
    }
    this.novel = novel
  }

  serialize () {
    return JSON.stringify(this.novel, null, 1)
  }
}
