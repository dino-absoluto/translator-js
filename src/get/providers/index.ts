/**
 * @file index.ts
 * @author Dino <dinoabsoluto+dev@gmail.com>
 * @license
 * This file is part of translator-js.
 *
 * translator-js is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * translator-js is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with translator-js.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
/* imports */
import { Provider, Novel } from './common'
/* code */
export { Provider, Novel, Chapter, Content, Formatter } from './common'

let initialized: Promise<void>
const domains = new Map()

export const register = (provider: Provider) => {
  for (const domain of provider.acceptDomains) {
    if (domains.has(domain)) {
      throw new Error(`'${domain}' domain existed`)
    }
    domains.set(domain, provider)
  }
}

export const getProvider = async (url: URL): Promise<Provider> => {
  if (!initialized) {
    initialized = (async () => {
      register((await import('./syosetu')).default)
    })()
    await initialized
    return getProvider(url)
  }
  let value = domains.get(url.hostname)
  if (!value) {
    throw new Error('No matching provider')
  }
  return value
}

export const getNovel = async (url: URL): Promise<Novel> => {
  const provider = await getProvider(url)
  let value = provider.fromURL(url)
  if (!value) {
    throw new Error('Failed to create Novel instance')
  }
  return value
}
