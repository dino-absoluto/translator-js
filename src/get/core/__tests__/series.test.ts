/**
 * @file Series
 * @author Dino <dinoabsoluto+dev@gmail.com>
 * @license
 * This file is part of translator-js.
 *
 * translator-js is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * translator-js is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with translator-js.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
/* imports */
import { Series } from '../series'
import { getNovel } from '../../providers'
import { back as nockBack, NockBackContext } from 'nock'
import del from 'del'
import * as path from 'path'
/* code */

const TMPDIR = path.resolve('__tmp__/jest-tmp/get/core-series')
const OUTDIR = path.join(TMPDIR, 'output')
nockBack.setMode('record')
nockBack.fixtures = path.resolve(TMPDIR, 'nock-fixtures/')

let nock: { nockDone: () => void; context: NockBackContext }

beforeAll(async () => {
  await del(path.join(OUTDIR, '*'))
  nock = (await nockBack('syosetu.json'))
})

afterAll(async () => {
  return nock.nockDone()
})

describe('Series', () => {
  const href = new URL('http://ncode.syosetu.com/n0537cm/')
  const title = '邪神アベレージ'
  test('constructor', async () => {
    const outputDir = path.join(OUTDIR, 'constructor')
    const series = new Series(await getNovel(href), {
      outputDir,
      data: {
        id: 'n0537cm',
        name: title
      }
    })
    expect(() => series.container.path).not.toThrow()
    await series.ready
    expect(series.data.name).toBe('邪神アベレージ')
    expect(series.container.path).toBe(path.join(outputDir, title))
  })
  test('serialize()', async () => {
    const outputDir = path.join(OUTDIR, 'serialize()')
    const series = new Series(await getNovel(href), {
      outputDir
    })
    await series.ready
    expect(series.data).toEqual(expect.objectContaining({
      id: 'n0537cm',
      over18: false
    }))
    expect(() => series.container.path).toThrow()
  })
  test('update()', async () => {
    const outputDir = path.join(OUTDIR, 'update()')
    const series = new Series(await getNovel(href), {
      outputDir
    })
    await series.ready
    expect(() => series.container.path).toThrow()
    await series.update()
    expect(series.data).toEqual(expect.objectContaining({
      name: '邪神アベレージ',
      author: '北瀬野ゆなき',
      genre: 'ハイファンタジー',
      status: { completed: true, size: 82 }
    }))
  })
})
