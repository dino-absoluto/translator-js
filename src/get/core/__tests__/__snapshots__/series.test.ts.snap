// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Series updateIndex() 1`] = `
Object {
  "邪神アベレージ/!meta/!cache.json": "cmlFaEhL7YTnv4clcN753F+GC2P97ItMFUvw6mbqr5QgsMY2KfA43gYFqbDFP2PI9leDxjipOKK8bdne32ejXQ==",
  "邪神アベレージ/001 【前篇～邪之章～】/001 01：些細な願い.txt": "8uHejteX1ZGrYY9KTSHu1MfJveZJRx1OmUftorAG1D9KtaDLfu9kpaIz8UyvFIdWjgYeqjum5JvzLu+E/GkXLQ==",
  "邪神アベレージ/001 【前篇～邪之章～】/002 02：平均的な結果.txt": "OMD/q8I+znwO1lko99JW7INRw4LTBbh9YgMnxtfBTgnOFQShQg5XyzMkXp+2cVDVjCsVQOxq9Z18viBCjPFF6g==",
  "邪神アベレージ/001 【前篇～邪之章～】/003 03：よくある出来事.txt": "y2460tX/+9Pbg9cNl8uXDoNenEGjk1YN6XL7eYQS0nS49OXRP6A18OtpoZFMr0xEk1D7ZNm38OTCvojVEDU2kw==",
  "邪神アベレージ/001 【前篇～邪之章～】/004 04：人心地.txt": "DYoY15ZYcvBl7XTI9ECE6/4vlxVlo2/q3UG6N6WNnJ3edBxE6txAOB6du5hXHXR5Jxq4W4GgqbPq4x60zDkLUQ==",
  "邪神アベレージ/001 【前篇～邪之章～】/005 05：聖なる場所.txt": "w80UFJ8ThpVnnDh9S2xDbPlHiaH20QuMv/edhg/awXm/f2SxwdkvjlBR+UX7/U+yJ6P7fS6pUYtOOrF9IqRFPQ==",
  "邪神アベレージ/001 【前篇～邪之章～】/006 06：お約束.txt": "aGI+Y4OlmPzOXD72JOy6Fl1jL3l6/NyVZM5ZGfVFNf3UBMnFAPsg4l3dOM6vSLW685lHXzkTS6i2PwwKT4wk/A==",
  "邪神アベレージ/001 【前篇～邪之章～】/007 07：転職？.txt": "Zk6DUR5iIRDapiRJ8MLkSZkTsd7+Slxd5E/VkRafhgLRFUWcAoa3ZcBppxj5TfAodKnOr24kRw9GGen4N4la7g==",
  "邪神アベレージ/001 【前篇～邪之章～】/008 08：お引越し.txt": "QtoF8tAFwdoW2lhnJudxViEOc43EIbPKaOEBpNAXGlJEqrwUCckzIECTDd7V8Ux1BE60dhmARZKLikEjN7iOEw==",
  "邪神アベレージ/001 【前篇～邪之章～】/009 09：はじめてのおかいもの.txt": "Dcs3bmzZX3rAHrbTopftdBx7FQxcgOspvSBw9MM2WMLEA8/CLDDyOINM9zpEOvebIlcynQDqugCXiQzWGV5fhA==",
  "邪神アベレージ/001 【前篇～邪之章～】/010 10：引き籠もり生活開始.txt": "sgRrjOmbwbmXq5quefzwCqDXixtXSfkjgh90WZJ2RgEFs9cETClksqGDOc/11oUSgyWccGRIbFq93mBzOaA/Sg==",
  "邪神アベレージ/001 【前篇～邪之章～】/011 11：注文の少ない迷宮.txt": "Ny/1NpDNeTv50CwUS/7qRzBW/BFqHf8/QoflrhpN/OdISMTgQ+Tw/9oWP0rc3ZMwEmRzs1l74EbkU/Ebk8mX3w==",
  "邪神アベレージ/001 【前篇～邪之章～】/012 12：邪神の追剥.txt": "6mlsK30H1/6aPvxRgF+gwXAHEP86d5wsqRuSakMPdm6qfWRcTdYHlxi3peLlxwktx/7Ulk8MVv6E+blserAy5g==",
  "邪神アベレージ/001 【前篇～邪之章～】/013 13：絶体絶命.txt": "1m2QLwqxWlWYG/qB04s/3n8nCkR6IBE8PJYSNnOD+8ZTQYtsThcRqSFI3OXNluJZ7pFDcQmvGwUIK6ZA+GUaPw==",
  "邪神アベレージ/001 【前篇～邪之章～】/014 14：天敵.txt": "4HooYFIGivX2m3zNDVzndqcmyvmlR/HEYpaH9nwPJZbEbg25Wa0ERxX1doRHSny7l8VDF9yzsPRMnUv9ze+Q2Q==",
  "邪神アベレージ/001 【前篇～邪之章～】/015 15：ラスボス戦.txt": "W0Kt3zajMWLjQKB4imYIKLutf2kCj6Xy1SAfQuaz2D1zcFvDBlrhvmvuom1X1a9r4OIGYWmZH+IQsqjIPN47zg==",
  "邪神アベレージ/001 【前篇～邪之章～】/016 16：賠償請求.txt": "Dzx+4l/96Drts5WM+HI4Fh9YpDumZ1T6Spl1GAeY4wYvZNk+3ydy2z/mGddzPt7OVVbOJRKdW6Ta1PleS+ANvg==",
  "邪神アベレージ/001 【前篇～邪之章～】/017 17：ガールズトーク.txt": "Z7I5/oCoJppkIvs3tE1Z+FGvlGdodnlenHvzM4Yr+55NFDQ/l71T5dI11AKPNgYE0oQpzdjge39wNxVTIxVXLQ==",
  "邪神アベレージ/001 【前篇～邪之章～】/018 18：狂宴.txt": "Z9vfMUpKgVD9Hy5q/KBfEymwr9mkh1l4xO0BjJyqLJpRBi6qQcUhi7DVk/S/k/XTuQCP1Wq+8MYvykrJfACI2A==",
  "邪神アベレージ/001 【前篇～邪之章～】/019 19：兵糧攻め.txt": "uPy2CqRILlwiUivy4cpGoUNTE1Y6kQcpmCiaMQtkZXmThFSDFfOQGqsoleHAJRQddV7Nmjc0DAHc4Ia8Iw4j/g==",
  "邪神アベレージ/001 【前篇～邪之章～】/020 20：神殿戦争.txt": "5SwE/cYuO+BWOrqFtFbdRwzwmDg5nMvD/vvOOo+w0C/2pMZoXVGg0bXqMZDoZsMwVa7ukZek0heMifTGFkPqMQ==",
  "邪神アベレージ/001 【前篇～邪之章～】/021 21：邪神アベレージ.txt": "N15LqC9TGvWpWjBdeJT4LkbsEcwXWOR7NykdqNk5ggqAoS2kAjRKZESfzmewLqbHy3mSidIiTIZWDyqBXbesDw==",
  "邪神アベレージ/002 【邪之章外伝】/022 外伝01：ある女将の接客.txt": "MH0clQ0MGEg0c/VBDahpf3vsBVuzhxFQxdY46HLVPvyZ2sxZk8a6LiVMYuYVtpeQeyKJf1DTRH5GTQO1AqwIiw==",
  "邪神アベレージ/002 【邪之章外伝】/023 外伝02：あるシスターの恐怖.txt": "/wvlAWTADBfdQOtT0XFUPgPR7Bod8hpjs7A40Uzhs7i9RTi/v3gQnKtUZFT0yteUnblWsq9ZEmObdoMRpMePdA==",
  "邪神アベレージ/002 【邪之章外伝】/024 外伝03：ある冒険者の災難.txt": "j0qmj5HyhuVFPYST/djbhXUrry8PwfIDcy3QANqJ2UQZ24JRj/wY7swWL1Q5L5IK1I2aQwQp/bz0O6s6e4jsjQ==",
  "邪神アベレージ/002 【邪之章外伝】/025 外伝04：ある奴隷の救済.txt": "RBJngW0lqnqYF0l5W7KPFY92FQB2RgneJgtQgrSJDMRdGyS+zefKWj8orZ1qu++NSYplRU2BMCtMqQyG/SIr7Q==",
  "邪神アベレージ/002 【邪之章外伝】/026 外伝05：あるギルドマスターの憂鬱.txt": "oekplAIMiXZxX3J6B7nNVIHsUmDZ2h0miybnyrsE0JMKRA0xsY574yagXCu9tP3RbaKFbLa+5DYrIAwfvNhu/w==",
  "邪神アベレージ/002 【邪之章外伝】/027 外伝06：ある魔導士の嘆息.txt": "U9cvKDJU3YkNHRh3kyl8rzhSWHpjEXTs/O9HXsg9QP1AHOmmWGd/69EfGQuY6ccqusV0sHML2pqYs7+snFIN2Q==",
  "邪神アベレージ/002 【邪之章外伝】/028 外伝07：ある不死王の忠誠.txt": "ydYo32VIVQvrc4rmo8osAG28FYhPtDFCobVECmDsv4wkI39QZRMtNRXWRgopwbPd3gVMQ/K4AC0sHnuQGBMiTA==",
  "邪神アベレージ/002 【邪之章外伝】/029 外伝08：ある魔王姫の交友.txt": "zCZQP5pVXC5guYCeZQfVVaeZxyhgVu9jqT7Xu4UhgAo194Xd/LGglDss8J4/xyxO1a7SmmyyfvQC3IcnuY7xaw==",
  "邪神アベレージ/002 【邪之章外伝】/030 外伝09：ある教主の信仰.txt": "id0NDFGw0L82fWMvtW14Jdaet39pQFdp+eFkAfuuZ2U9Xor6XVBiwR0eTjAZpf41o63RZw7mUY7kqLDHY7Qkag==",
  "邪神アベレージ/002 【邪之章外伝】/031 外伝10：ある王子の絶望.txt": "RI+G9fluTksS+UpsUEjhB+ZcZ+NvamoTOvFItW42xwNCsIO1rkwm4MWRhp9Zcg0FCof/US673ahoseF6m9wkOg==",
  "邪神アベレージ/002 【邪之章外伝】/032 外伝11：ある邪神の嘲笑.txt": "E9LsDlfhUua1LaO1L9aUG3FplAC3r8RFQ53IMjDxexaZYQ5eLD80zxkVaLXkdIc2emFa9q9BKRIXux5DRIue8g==",
  "邪神アベレージ/003 【後篇～神之章～】/033 01：黒き歴史の始まり.txt": "hosaCE3v2lrEouTrH/cVN+75wa3g+kdTsk43Gx9br5lJJ7LnRjIK0DkSoKrtOV6Cbs/0G6FceMoEGp7x30cutQ==",
  "邪神アベレージ/003 【後篇～神之章～】/034 02：各国模様.txt": "w5H15dJBwUM9ljpUzo5Aa7i7yaQjH32Ov/5yitDQKL+fM7MdQZjEHKQJaK1wRRdFBUYkxC+U3dVXtKZNwGbB0A==",
  "邪神アベレージ/003 【後篇～神之章～】/035 03：邪教のすゝめ.txt": "TV/d/E8o1ROh2xxzrUE8s0zQAV3GBVmcZW8ZvNJqCLaHBBCpwUNmLMdRS6Y5bWMEh0pE17JGUIQXtOZY93yRCw==",
  "邪神アベレージ/003 【後篇～神之章～】/036 04：光と闇.txt": "U7HGg7sC+IpRwbDCZOPHko9XF3S/BkJuJtkAoI9r1Go5Pu/JICvPWFTA6UXyNUnICe08BtULipybCdhd7p07jg==",
  "邪神アベレージ/003 【後篇～神之章～】/037 05：会議は踊る.txt": "5WAOM2S52ClZVXqlQvqaPWfjUyFr/XltcuCxZDSxfY0b6KPAhmW3EFfn0enN2JjWITCgXWDCoWlH7cGN2vGxng==",
  "邪神アベレージ/003 【後篇～神之章～】/038 06：その日世界が震撼した.txt": "RPjCyak05tH3ucueyjXGv86/iwwcIVegVQvcABBL836XlSr164jP9vGG8NUBeX8SHtDWKZArWsgtBnTUyu5oxQ==",
  "邪神アベレージ/003 【後篇～神之章～】/039 07：ダンジョン再始動.txt": "RSKUl7D/IPIme7zT9LkPsx51IPYumlHk5ictn1JtdmKe2TfIpXerBv+pyvZbsty7HIpRzxFt0B/D2ANH/EaZmQ==",
  "邪神アベレージ/003 【後篇～神之章～】/040 08：侵入者改めお客様.txt": "p+MZ1x3+SfTuCcyyf0HFJN2rjQPWGVuW5uZDMEOz0K22EuEyAkoEdqfr/yiC8PSNTEJQK/S0fnKa9MMgiVtURA==",
  "邪神アベレージ/003 【後篇～神之章～】/041 09：彼らが帰ってきた.txt": "CnslPHO6pfZOhPD9tPJodiN4U44XatglqKKNN1zXYX9f3j9aDqJAcVEKq/y6NIOmDiZA8vk0ERHuh0re/aA1Ig==",
  "邪神アベレージ/003 【後篇～神之章～】/042 10：黒き暴君.txt": "IvomS/SWHe1Pm4VeU8kVbOl7HzQlqui1Z2nIqZuAIX0iPt1N3tF3g6KBJaCxzRxWylyCztjdZwKBIU8iBb6vXw==",
  "邪神アベレージ/003 【後篇～神之章～】/043 11：四の天に座す王.txt": "zoDLo4FwWoifEn/6SDScxkdX1ax5keJZUTCWTG9OaMPd+5fmliehgPccT0W36VGwyTlp8hLpTCUSVj3OiaiQbQ==",
  "邪神アベレージ/003 【後篇～神之章～】/044 12：嵐の前の一時.txt": "Z4IOZfUA24M7AJ45x5mUBKSFFMHPuuSyhBc93IM7bWjAhXLpVhOm+t8B2bBZcmOKSkvTueyDf2MliwpFoiqTqw==",
  "邪神アベレージ/003 【後篇～神之章～】/045 13：大人げない.txt": "P6d1W6DHn0c+dAv5T6PBJcjFNZSmroMq/AS0RceHPVZOWKDGFiNMFPlq5lsMur15MxMGcgqA0m1wHmR+kkgWPg==",
  "邪神アベレージ/003 【後篇～神之章～】/046 14：謎解き地獄.txt": "N6DA9DB5Aa7yoYaFNAjgOXcKMFIHQ5WNQwDqT2mYRC2shYS3bFr33oEtgMg9MNlLsxZWF0CafpPjG5bH5Mte5Q==",
  "邪神アベレージ/003 【後篇～神之章～】/047 15：邪神の鎧.txt": "8PXyq+7rAv/GP9Apd+TOh3m4KQuUV3ckxNTP0pVID2SoUHcw1h99YbCnPkJdTcfK1u1TvX3zNWCJ0jOMjto6GA==",
  "邪神アベレージ/003 【後篇～神之章～】/048 16：ボスラッシュ.txt": "QLtt7yWjALu6NNKkSXIsdd8NESUGwfaTXMs06ZzRg9oGsnG7+74BuuBMls565DH2yHQ7hrzKImbYVElXeaBKcA==",
  "邪神アベレージ/003 【後篇～神之章～】/049 17：インペリアル・デス.txt": "5IRn4a49DcgsKNWte/+T7135JWXxAcKth8qJN666elnIymRE6E8pGtGv+ebZhXYR+ln+tBFGNZ4ATOPzYqyCqg==",
  "邪神アベレージ/003 【後篇～神之章～】/050 18：決着.txt": "LuWK7FzqR59ST172mcc0ADcyAMkcLIGnm++9FvrckARPRqXYYzxeInKYkUQUpWkodgTviHXTSXzoLTbE/7Ew1g==",
  "邪神アベレージ/003 【後篇～神之章～】/051 19：邪神の誘惑.txt": "OB1t8M7lzGv/o1vHdfPUZKaXCaaKj6GKXfxgMweP7MY3yMaREdad+UzRg8EAAW29e0NPcKe59496uPJ+QkiWuQ==",
  "邪神アベレージ/003 【後篇～神之章～】/052 20：普通のアンリ.txt": "i1p0pLdovsWPx8zRyNpCzH2/H5GzG/hUZClC6Bu6FHahk1AWVNvbrbpyzQCu5l5mxrCvIaBz33LbjMhLAvD2Lw==",
  "邪神アベレージ/004 【神之章外伝】/053 外伝01：ある宗教の転落模様.txt": "u3cY/fXGIKJvayHoC3L8LjjqydYbqYDwYOgtVZNomLdsp+kaz/MYcD5wy++v3zR7tzj4odz5rwnfbdxpSz44sw==",
  "邪神アベレージ/004 【神之章外伝】/054 外伝02：ある経典のパンデミック.txt": "dEZXR/8LcYI+KPH8U7ukj0f7qKlJqyFsQ59L82ywXp37r8IF1HmSEMSrSouV9y6Z3lV/RwPQj7Bc4fxSA+6BXA==",
  "邪神アベレージ/004 【神之章外伝】/055 外伝03：ある主従のおさんぽ.txt": "Bu/jWmdgBzKjs2f94YAZM9QGAeov2F+Wc0jNHJeEKlULOufxM52WTGrIhMHlnNR55Dyg1jpNFILC7wEcnDhHSg==",
  "邪神アベレージ/004 【神之章外伝】/056 外伝04：ある勇者達の労働.txt": "t+wMQD9Qy7uFRZsEJ/Cl+iQ38l7ZTHza9GQ4uoXcYcHcMIPbSAZHswPxNvAW8JMj/iEX1n/TI3bQyUMgy93HCA==",
  "邪神アベレージ/004 【神之章外伝】/057 外伝05：ある従者の里帰り.txt": "KbrfG4YJOCF9MawIzT7yQ+z5t59AttX7ZlB7bXciDvBeB9LouuLsqbc6pqDGfRzPqG3uFveB3oXn+/X3kvV0kQ==",
  "邪神アベレージ/004 【神之章外伝】/058 外伝06：ある魔王の会食.txt": "Nd6vKp78tYHe6vvK+AFsFNel4N6zJXLvFFz6k3oybTonja8n4ixBNyAC0/VtWuMsCrmEpBSadRUdJU8peLlTnw==",
  "邪神アベレージ/004 【神之章外伝】/059 外伝07：ある教皇の企み.txt": "HQCxqoGCmBQ7rkGyde5ZpreuDr2HAB4vvi6eN0ZEFQ3NWvgiQg46EuB87rMBSmXmcJsoQ1YMVX0wv9Mauld80Q==",
  "邪神アベレージ/004 【神之章外伝】/060 外伝08：ある不死のバトラー.txt": "a/ijj4IPYanIibAA8Pixd6Ys3Fnr8RCdTji1zMHvbDfd5gM4fg1eDc3iNwZv80t/zxB0E5Ge9hiA/GFolRPiiQ==",
  "邪神アベレージ/005 【外の理】/061 記念SS：自伝「アンリと愉快な仲間達」.txt": "kW7/xJySnK1fwkBUrkaRK4zIDe43c77vCB0gdYObGlbYxjCew0kE/6aoxSHLY/rYpqBWVXY1ftSbPWAH0D3XTg==",
  "邪神アベレージ/005 【外の理】/062 記念SS：まんぷくアンリさん.txt": "KenGAc4b3XUnzjKRQiaoJrZ8mLHJPaVg2sZIjuxM6umQ86CrINLaoDgfDLm7Lq7IuCjj47vKU+PF7jEltKRuSg==",
  "邪神アベレージ/005 【外の理】/063 記念SS：異伝「クリスマス禁止令」.txt": "CWm4s2OuP+XetR3Ue/yA8h2JjIFkxZuQdc/JCTDSFnBAJXVEg3xjxboYmkzG4un1GI7Dcyukfs0xSNqRXqRHcw==",
  "邪神アベレージ/005 【外の理】/064 記念SS：謎伝「明けまして同文化交流」.txt": "6IDRsd6Y8ZlsCd7mtnVJ5ZHyy0vujLX9KxpCBTwk6Jo+qktAFBsUGC1LIFKZbVHGlUnYLIYWYJxaGdmYlSBqWQ==",
  "邪神アベレージ/005 【外の理】/065 記念SS：他伝「崇高なる我が神」.txt": "5j+6cXvrzkHZ1cRXlrgPCvxjnUD9XJn29aKR/5OtTUoM1bn67bm80eVXD7Cyw0NyAR3uRbN7/iUzdNRApRsr7A==",
  "邪神アベレージ/006 【付篇～平均之章～】/066 01：危険な邪神像.txt": "xxgfVDsL2PiPMu/ZHBGLmh4PkkYXXtFP0m+uHCIxsvx0ccrsXcI+4Xgegww8Imv5gEyZiihpKVKqOgWVBW6Pjg==",
  "邪神アベレージ/006 【付篇～平均之章～】/067 02：邪神像起動！.txt": "TbeVqJMbbckZY2sXTUY/sLvJZn/iZfuicsab4o4KyzSFSO7a28qiEV6nfbwTgnv8cYcReZr3r08vdy496/sVOQ==",
  "邪神アベレージ/006 【付篇～平均之章～】/068 03：蹂躙される世界.txt": "ZoLmfnopWOFIL0UDnayV+ytZ4BkKXQioECvVrq3XErsQmC/YoHWc1r7Hk2CVKWGk8yx5B+YPSPwizgBsT0CE4Q==",
  "邪神アベレージ/006 【付篇～平均之章～】/069 04：三柱査問会.txt": "6knQPLAZZNOyeS0XCMls3lk3pt085p+vpbjWZLzzF8vBujs1UAF2IqTfU7Za4yGFi/FUuQ4RlGXIgUqxUQkMFg==",
  "邪神アベレージ/006 【付篇～平均之章～】/070 05：選抜メンバー.txt": "6qOjleGpaxnTghl/GgyKA0Ez6TPnKCn3HL7ossgIp8pK4NlbMRqhGPftYWrcbm867SbT1sd38qs5PQXpPRQkAw==",
  "邪神アベレージ/006 【付篇～平均之章～】/071 06：対策会議.txt": "nnzlLe7ROuTt/v9G3LVNX35PHcCllRJmuReeQ4xwKNB1Gw1da1wHo9vha/Ya+5j8UQXlEmS6ALIxGZJmREKlJg==",
  "邪神アベレージ/006 【付篇～平均之章～】/072 07：トラップ.txt": "7RdNb7QuCqYu/28GItGQAfx8kd3kfFgV6wff3V86+S+XzS/WMBCcRwUApJWmIi6dpevCBYlxxK4DZ3R1BS5/5A==",
  "邪神アベレージ/006 【付篇～平均之章～】/073 08：撤退.txt": "AnHFQlKRvkntpGWMKhFw6R42YG1k6aQVcPnwwzwQ+X0Oeh43Q20C4YadPDRw+inRtiVbRC6O7PyVtZwIhyBWIA==",
  "邪神アベレージ/006 【付篇～平均之章～】/074 09：対策会議、再び.txt": "MzGlx+4GNXsWjYMj7P9anmwNF7eM2rooEk0rBMt6SFJjhbGGzIejEVBvleWkISbi09BHyjJQFoH6ThWtBMoV3A==",
  "邪神アベレージ/006 【付篇～平均之章～】/075 10：大切なもの.txt": "iLpotD9jzHchkIDu4dv3S2KFu87UY7SGd2MzBHLKAMgwY+D+Y1AIcKJe/G80DMOWJgLadvouvty313TPLu4BHw==",
  "邪神アベレージ/006 【付篇～平均之章～】/076 11：封印.txt": "ZI1yl/zF8Q6xOnQp26QFmSkv2k6O+ptwhzaUNU4h+fPhp28U5dP0UVny6c7X34dnQhyjnBHZxSPCCBoqLTtL0g==",
  "邪神アベレージ/006 【付篇～平均之章～】/077 12：後始末.txt": "P4zQlrCFrCBHfBZvUImlNz7tgU8+Agf28MU1qO7KC6MEooNcfQ91xAjIcPVGq3z7ph8Ts3c/OHVf1ra1S2FIIg==",
  "邪神アベレージ/006 【付篇～平均之章～】/078 13：戦慄の結末.txt": "35p/iBydUaEFHIdOQDZc7OR8jPT2wQpRZNeFsOyrx+lMV7P/r/mhGZ9vU0Wvrew26Hzm1Dcts1W/oiOp5iPqnQ==",
  "邪神アベレージ/007 【特典之章】/079 アンリ様のダイエット.txt": "IeCpgT4ZLQywGWvFdpKdJRtOCiCzHPAQGheMvPun+4UhkqvK369STBvEfFSgRry80yz9A32XZ/tzUqx2xtzHvA==",
  "邪神アベレージ/007 【特典之章】/080 テナとカルチャーギャップ.txt": "m8AGmSX9ezk30f4IeEtJuVcLIBfOt5p6pgfMCiCj89yrv1YcJ1DBcS9n/GE1L9W0BMyLMiZVM7JyTmRlC1iwrQ==",
  "邪神アベレージ/007 【特典之章】/081 教えてレオノーラ先生！.txt": "Y3F6xqQFwlGHiqSWRa7qwWcRetyBpb1X7wzE+9T7cHPgKGeaINoYNZCkuEP0838ohyeEA2Z2NPOD5Lw3yiAM4g==",
  "邪神アベレージ/007 【特典之章】/082 不死王様の退屈しのぎ.txt": "AhII/6jARcuNZCfDJZIdRhHWX41ak7iNXXZsJRUqReSyE7PBj9biWG9QCZ5HeINoWHftzZbksOVT9rxqhZlHNQ==",
  "邪神アベレージ/index.json": "vnMuQcNM6uvf3nmH1IgRIH/fzTU/469n70oCCjjEX6owndlNhLaG4xVtNL0QOJpcIeW8E0qw6eAlLldbklpClw==",
}
`;
